lib/core/src/Cardano/Wallet.hs

```diff
@@ -352,18 +355,22 @@ cancelWorker (WorkerRegistry mvar) wid =
                                  Construction
 -------------------------------------------------------------------------------}
 
+newtype SlotLength = SlotLength (Quantity "second/slot" Word8)
+    deriving (Show, Eq)
+
 -- | Create a new instance of the wallet layer.
 newWalletLayer
     :: forall s t. (Buildable (Tx t))
     => Trace IO Text
     -> Block (Tx t)
         -- ^ Very first block
     -> FeePolicy
+    -> SlotLength
```

Comment:
I think it'd make sense to introduce an intermediate type now to factor out some of the parameters. Something like:

```hs
BlockchainParameters t
    { genesisBlock :: Block (Tx t)
    , feePolicy :: FeePolicy
    , slotLength :: SlotLength
    }
```

Reply:
What do you think ?

Reply:
make sense